src/Readers/Vipr.Reader.OData.v4/OdcmReader.cs

```diff
@@ -314,7 +314,8 @@ private void WriteNamespaceDeep(IEdmModel edmModel, string @namespace)
                     }
                 }
 
-                var entityTypes = AllEntityTypes(types).ToList();
+                // Order by elements without parent(Base Types) first
+                var entityTypes = AllEntityTypes(types).OrderBy(static type => type.BaseType != null).ToList();

```

Comment:
we should probably add a then by statement using the name as well to get consistent output even with mixed order input